### PR TITLE
Create/Update Major.Minor specific CHANGELOG files.

### LINKED ISSUE #427: changelog generation should write to file per release-branch
In order to keep changelog files from growing without bound, and to make it easier to see what is changing in a given release stream, the changelog files are being split per release branch.

However, anago is still generating new point release notes in the main CHANGELOG file. It should target release notes for tag `x.y[...]` to `CHANGELOG-x.y.md` (creating it if needed)

### PR DESCRIPTION
**Pull request by david-mcmahon**
Fixes https://github.com/kubernetes/kubernetes/issues/48985
Fixes #427

### CODE REVIEW

File: anago [hunk 1]
```diff
@@ -403,25 +403,41 @@ generate_release_notes() {
...
-  logecho -n "Fetch origin/master to get latest CHANGELOG.md: "
+  logecho -n "Fetch origin/master to get latest $CHANGELOG_FILE: "
   logrun -s git fetch origin master || return 1
-  logecho -n "Rebase on origin/master before editing CHANGELOG.md: "
+  logecho -n "Rebase on origin/master before editing $CHANGELOG_FILE: "
   logrun -s git rebase origin/master || return 1
 
-  logecho "Insert $RELEASE_VERSION_PRIME notes into CHANGELOG.md..."
+
+# TODO: Is there any value/risk/benefit to creating a symlink to CHANGELOG.md
```
Comment (xiangpengzhao) [inline anago]: Users may be a bit confused at the first sight when they see CHANGELOG.md and CHANGELOG-1.NewestVersion.md are both the changelog of the newest version ?

Comment (liggitt) [reply anago]: aliasing CHANGELOG.md to something unstable leads to links that break... I'd prefer just the release-branch-specific files

File: changelog-update [hunk 2]
```diff
@@ -22,15 +22,13 @@ PROG=${0##*/}
...
 #+
 #+ SYNOPSIS
-#+     $PROG  <tag|all> [<tag ...]
+#+     $PROG  <tag> [<tag> ...]
 #+     $PROG  [--helpshort|--usage|-?]
 #+     $PROG  [--help|-man]
 #+
 #+ DESCRIPTION
-#+     Look in kubernetes/CHANGELOG.md for <tag>'s and update the release
+#+     Look in kubernetes/CHANGELOG1.N.md for <tag>'s and update the release
```
Comment (liggitt) [inline changelog-update]: `CHANGELOG-x.y.md`

File: docs/release-notes.md [hunk 3]
```diff
@@ -5,30 +5,30 @@ blocks)](https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUE
 labels](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes) with compliance across master and release- branches using the munger in [contrib/mungegithub](https://github.com/kubernetes/contrib/blob/master/mungegithub/README.md#submit-queue)
 on the main Kubernetes repository (and future repositories later).
 
-Releases are built and published by the anago tool in [this repo](https://github.com/kubernetes/release) with the release notes published in [kubernetes/CHANGELOG.md](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md).
+Releases are built and published by the anago tool in [this repo](https://github.com/kubernetes/release) with the release notes published in [kubernetes/CHANGELOG1-N.md files](https://github.com/kubernetes/kubernetes).
```
Comment (liggitt) [inline docs/release-notes.md]: `CHANGELOG-x.y.md`

File: relnotes [hunk 4]
```diff
@@ -55,7 +55,7 @@ PROG=${0##*/}
 #+ OPTIONS
 #+     --quiet                   - Don't display the notes when done
 #+     --htmlize-md              - Output markdown with html for PRs and
-#+                                 contributors (for use in CHANGELOG.md)
+#+                                 contributors (for use in CHANGELOG1.N.md)
```
Comment (liggitt) [inline relnotes]: `CHANGELOG-1.N`

**Comment by david-mcmahon** [comment]
PTAL

**Pull request merged**